azure-kusto-data/src/arrow.rs

```diff
@@ -192,16 +154,19 @@ pub fn convert_table(table: DataTable) -> RecordBatch {
             .for_each(|(idx, value)| buffer[idx].push(value))
     });
 
-    buffer
+    let iter = buffer
         .into_iter()
         .zip(table.columns.into_iter())
-        .map(|(data, column)| convert_column(data, column))
-        .for_each(|(field, array)| {
-            fields.push(field);
-            columns.push(array);
-        });
+        .map(|(data, column)| convert_column(data, column));
 
-    RecordBatch::try_new(Arc::new(Schema::new(fields)), columns).unwrap()
+    for result in iter {
```

Comment:
This is something where I have a bit of an opaque understanding. I was once "told" the when we use the functional like APIs (`for_each`) that Rust can do more optimizations internally, specifically vectorization. I guess in this context it's not going to matter much anyhow, but just wondering if you know more about this?

Reply:
I remember this being true on the "more functional" methods (map, filter, reduce), but I'm not sure the extent.
I changed it because I needed to unwrap the result, but now I discovered `try_for_each` so we can keep it :)